components/MapComponent.tsx

```diff
@@ -23,83 +32,142 @@ interface Props<T> {
     longitudeDelta: number
   }
 
-  markers: mapMarker[]
+  initialRadius: {
+    radius: number
+  }
+
+  markers: MapMarker[]
+  customizers: MarkerCustomize[]
+
+  currentSelectedMarker: String | undefined
 
   movementSettings: {
     canScroll: boolean
     canZoom: boolean
     canRotate: boolean
   }
+}
 
-  extractKey: (event: any) => string
+// When clicking on the marker, zoom towards where it is.
+// Using stackOverflow formula
+// stackoverflow.com/questions/5206018/fit-mapview-to-circle-bounds
```

Comment:
The link takes me to some weird 2011 post about setting up reliable UDP sockets in C 😆 (which isn't possible, since UDP is really just the IP protocol in the transport layer)...